x-pack/plugins/apm/server/lib/services/annotations/get_derived_service_annotations.ts

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { isNumber } from 'lodash';
+import { Annotation, AnnotationType } from '../../../../common/annotations';
+import { SetupTimeRange, Setup } from '../../helpers/setup_request';
+import { ESFilter } from '../../../../typings/elasticsearch';
+import { rangeFilter } from '../../helpers/range_filter';
+import {
+  PROCESSOR_EVENT,
+  SERVICE_NAME,
+  SERVICE_ENVIRONMENT,
+  SERVICE_VERSION
+} from '../../../../common/elasticsearch_fieldnames';
+
+export async function getDerivedServiceAnnotations({
+  setup,
+  serviceName,
+  environment
+}: {
+  serviceName: string;
+  environment?: string;
+  setup: Setup & SetupTimeRange;
+}) {
+  const { start, end, client, indices } = setup;
+
+  const filter: ESFilter[] = [
+    { term: { [PROCESSOR_EVENT]: 'transaction' } },
+    { range: rangeFilter(start, end) },
+    { term: { [SERVICE_NAME]: serviceName } }
+  ];
+
+  if (environment) {
+    filter.push({ term: { [SERVICE_ENVIRONMENT]: environment } });
+  }
+
+  const versions =
+    (
+      await client.search({
+        index: indices['apm_oss.transactionIndices'],
+        body: {
+          size: 0,
+          track_total_hits: false,
```

Comment:
This should be the default so should not matter

Reply:
The default is `10000` rather than `false`. See the responses for not setting it and setting it to `false`:

```json
{
  "took" : 0,
  "timed_out" : false,
  "_shards" : {
    "total" : 1,
    "successful" : 1,
    "skipped" : 0,
    "failed" : 0
  },
  "hits" : {
    "total" : {
      "value" : 10000,
      "relation" : "gte"
    },
    "max_score" : null,
    "hits" : [ ]
  }
}
```

vs

```json
{
  "took" : 0,
  "timed_out" : false,
  "_shards" : {
    "total" : 1,
    "successful" : 1,
    "skipped" : 0,
    "failed" : 0
  },
  "hits" : {
    "max_score" : null,
    "hits" : [ ]
  }
}
```

I'll remove it though, it won't make any significant difference in terms of performance.

Reply:
ah, didn't think about that. Good point 👍